coding_systems/versioning/models.py

```diff
@@ -2,39 +2,38 @@
 from django.conf import settings
 from django.db import DEFAULT_DB_ALIAS, connections, models
 from django.db.migrations.executor import MigrationExecutor
-
-from opencodelists.hash_utils import hash, unhash
+from django.utils.functional import cached_property
+from django.utils.text import slugify
 
 
 class CodingSystemVersionManager(models.Manager):
-    def get_by_hash(self, hash):
-        """Return the CodingSystemVersion with given hash."""
-        id = unhash(hash, "CodingSystemVersion")
-        return self.get(id=id)
-
     def most_recent(self, coding_system):
         return self.filter(coding_system=coding_system).order_by("-valid_from").first()
 
 
 class CodingSystemVersion(models.Model):
     coding_system = models.CharField(max_length=10)
-    version = models.CharField(max_length=255, null=True, blank=True)
+    version = models.CharField(max_length=255)
     valid_from = models.DateField()
     import_timestamp = models.DateTimeField(auto_now_add=True)
     import_ref = models.TextField()
+    slug = models.SlugField(max_length=255, unique=True)
 
     objects = CodingSystemVersionManager()
 
     class Meta:
-        unique_together = ("coding_system", "valid_from")
-
-    @property
-    def hash(self):
-        return hash(self.id, "CodingSystemVersion")
+        unique_together = ("coding_system", "version", "valid_from")
 
-    @property
+    @cached_property
     def db_name(self):
-        return f"{self.coding_system}_{self.hash}"
+        return self.slug
```

Comment:
Do we need both `db_name` and `slug`?

Reply:
Originally I just called the SlugField "db_name", but then I changed my mind and thought it was more intuitive to name it "slug" and alias it to "db_name".  There were places where I was retrieving the `CodingSystemVersion` and it didn't feel quite right to be retrieving it by db_name.  I'm not committed to it though

Reply:
I've no strong opinion.